scrapy/utils/reactor.py

```diff
@@ -114,8 +131,10 @@ def set_asyncio_event_loop(event_loop_path: str | None) -> AbstractEventLoop:
     """Sets and returns the event loop with specified import path."""
     if event_loop_path is not None:
         event_loop_class: type[AbstractEventLoop] = load_object(event_loop_path)
-        event_loop = event_loop_class()
-        asyncio.set_event_loop(event_loop)
+        event_loop = _get_asyncio_event_loop()
+        if not isinstance(event_loop, event_loop_class):
+            event_loop = event_loop_class()
+            asyncio.set_event_loop(event_loop)
```

Comment:
@wRAR I don’t fully understand why this fixed [the hang](https://github.com/scrapy/scrapy/pull/6729#issuecomment-2755790919) or, more importantly, why this PR triggered the hang in the first place.